crates/engine/src/prepare/operation/mod.rs

```diff
@@ -94,3 +102,17 @@ fn mutation_not_allowed_with_safe_method<OnOperationResponseHookOutput>() -> Res
         )],
     )
 }
+
+fn validate_prepared_operation(operation: &PreparedOperation, settings: &Settings) -> PrepareResult<()> {
+    let ComplexityControl::Enforce { limit, .. } = settings.complexity_control else {
+        return Ok(());
+    };
+    let Some(complexity) = operation.complexity else {
+        return Ok(());
```

Comment:
Shouldn't that be an `InternalError` or smth?

Reply:
Although I should at the least add an error message in probably.

Reply:
yeah but silently having a problem sounds worse to me. At least there should be a `tracing::error!` in that case